torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -2189,6 +2191,65 @@ RegisterOperators reg2({
     Operator("aten::hash(float t) -> int", hashValue<double>),
 });
 
+bool simpleClassTypeArg(const Argument& arg, const ClassTypePtr& type) {
+  return arg.type() == type && !arg.kwarg_only() && !arg.default_value();
+}
+
+void checkSortInput(const Node* node, const c10::TypePtr& list_element_type) {
+  if (auto class_type = list_element_type->cast<ClassType>()) {
+    if (auto method = class_type->getMethod("__lt__")) {
+      const auto& lt_schema = method->getSchema();
+      const auto& schema_args = lt_schema.arguments();
+      bool error = schema_args.size() != 2;
+      if (!error) {
+        error |= !simpleClassTypeArg(schema_args[0], class_type) ||
+            !simpleClassTypeArg(schema_args[1], class_type);
+      }
+      error |= lt_schema.returns().size() != 1 ||
+          lt_schema.returns()[0].type() != BoolType::get();
+      if (!error) {
+        return;
+      }
+    }
+  }
+
+  auto error_msg = script::ErrorReport(node->getSourceLocation());
+  error_msg << "Input to list sort must be of Tensors, ints, floats, bools or "
+            << "a User Defined Class that defines the __lt__ compare method"
+            << ", got list of " << list_element_type->str() << "\n";
```

Comment:
I agree. The logic is confusing (using bitwise or on a boolean). This can all be put into a single if statement:

```
if (schema_args.size() != 2 || 
    lt_schema.returns().size() != 1 || ...
```